System/Process.hs

```diff
@@ -870,6 +869,22 @@ It will therefore behave more portably between operating systems than 'system'.
 The return codes and possible failures are the same as for 'system'.
 -}
 rawSystem :: String -> [String] -> IO ExitCode
-rawSystem cmd args = do
-  (_,_,_,p) <- createProcess_ "rawSystem" (proc cmd args) { delegate_ctlc = True }
+rawSystem cmd args = executeAndWait "rawSystem" (proc cmd args) { delegate_ctlc = True }
+
+-- ---------------------------------------------------------------------------
+-- executeAndWait
+
+-- | Create a new process and wait for it's termination.
```

Comment:
Given the change in behavior, and the fact that we cannot replicate the same behavior on POSIX systems as we do on Windows, I can't imagine a way in which we can add this new functionality to the default `createProcess` settings without wide-spread breakage. We _could_ add a new flag to `CreateProcess` to indicate if this logic should be used. However, this behavior is different enough, and applied to a narrow enough use case (`exec` usage on Windows) that I'd probably stick with a separate function and make very clear the narrow window\* for which this is intended and the caveats that apply.

\* No pun intended